src/tbb_concurrent_queue_scheduler.cpp

```diff
@@ -231,7 +233,7 @@ namespace CnC {
 					m_queues[_aff].push( stepInstance );
 				}
             } else {
-                m_gQueue->push( stepInstance );

```

Comment:
I don't understand the question.
There is a mqueue->try_pop before the blocking pop. Whenever a step is put and at least one thread is blocking we also put a wakeup step.